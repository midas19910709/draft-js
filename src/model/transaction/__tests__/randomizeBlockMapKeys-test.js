/**
 * Copyright (c) 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @emails oncall+ui_infra
 * @format
 */

'use strict';

jest.disableAutomock();

jest.mock('generateRandomKey');

const BlockMapBuilder = require('BlockMapBuilder');
const ContentBlock = require('ContentBlock');
const ContentBlockNode = require('ContentBlockNode');
const Immutable = require('immutable');

const randomizeBlockMapKeys = require('randomizeBlockMapKeys');

const {List} = Immutable;

const assertRandomizeBlockMapKeys = blockMapArray => {
  expect(
    randomizeBlockMapKeys(
      BlockMapBuilder.createFromArray(blockMapArray),
    ).toJS(),
  ).toMatchSnapshot();
};

beforeEach(() => {
  jest.resetModules();
});

test('must be able to randomize keys for ContentBlocks BlockMap', () => {
  assertRandomizeBlockMapKeys([
    new ContentBlock({
      key: 'A',
      text: 'Alpha',
    }),
    new ContentBlock({
      key: 'B',
      text: 'Beta',
    }),
    new ContentBlock({
      key: 'C',
      text: 'Charlie',
    }),
    new ContentBlock({
      key: 'D',
      text: 'Delta',
    }),
  ]);
});

test('must be able to randomize keys for ContentBlockNodes BlockMap and update reference links to the new keys', () => {
  assertRandomizeBlockMapKeys([
    new ContentBlockNode({
      key: 'A',
      text: '',
      children: List(['B', 'D']),
    }),
    new ContentBlockNode({
      key: 'B',
      parent: 'A',
      children: List(['C']),
      nextSibling: 'D',
      text: '',
    }),
    new ContentBlockNode({
      key: 'C',
      parent: 'B',
      text: 'X',
    }),
    new ContentBlockNode({
      key: 'D',
      parent: 'A',
      prevSibling: 'B',
      text: 'Y',
    }),
  ]);
});
