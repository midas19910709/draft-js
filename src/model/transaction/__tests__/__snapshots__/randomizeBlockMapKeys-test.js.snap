// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must be able to randomize keys for ContentBlockNodes BlockMap and update reference links to the new keys 1`] = `
Object {
  "key0": Object {
    "characterList": Array [],
    "children": Array [
      "key1",
      "key3",
    ],
    "data": Object {},
    "depth": 0,
    "key": "key0",
    "nextSibling": null,
    "parent": null,
    "prevSibling": null,
    "text": "",
    "type": "unstyled",
  },
  "key1": Object {
    "characterList": Array [],
    "children": Array [
      "key2",
    ],
    "data": Object {},
    "depth": 0,
    "key": "key1",
    "nextSibling": "key3",
    "parent": "key0",
    "prevSibling": null,
    "text": "",
    "type": "unstyled",
  },
  "key2": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "key2",
    "nextSibling": null,
    "parent": "key1",
    "prevSibling": null,
    "text": "X",
    "type": "unstyled",
  },
  "key3": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "key3",
    "nextSibling": null,
    "parent": "key0",
    "prevSibling": "key1",
    "text": "Y",
    "type": "unstyled",
  },
}
`;

exports[`must be able to randomize keys for ContentBlocks BlockMap 1`] = `
Object {
  "key0": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "key0",
    "text": "Alpha",
    "type": "unstyled",
  },
  "key1": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "key1",
    "text": "Beta",
    "type": "unstyled",
  },
  "key2": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "key2",
    "text": "Charlie",
    "type": "unstyled",
  },
  "key3": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "key3",
    "text": "Delta",
    "type": "unstyled",
  },
}
`;
